### PR TITLE
update deps of libzauth

### PR DESCRIPTION
**Pull request by jschaul**
Upgrade version of libzauth dependencies, notably sodiumoxide bindings to libsodium, and fix resulting errors and warnings.

libzauth is used as part of the nginz module for authorization.

This PR was triggered by dependabot warnings.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: libs/libzauth/libzauth-c/Cargo.lock [hunk 1]
```diff
@@ -1,71 +1,71 @@
...
 [[package]]
 name = "serde"
-version = "1.0.66"
+version = "1.0.136"
 source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "ce31e24b01e1e524df96f1c2fdd054405f8d7376249a5110886fb4b658484789"
 
 [[package]]
 name = "sodiumoxide"
 version = "0.0.16"
```
Comment (stephen-smith) [inline libs/libzauth/libzauth-c/Cargo.lock:47]: https://github.com/wireapp/wire-server/security/dependabot/1 says we need to update this to 0.2.5 or better...

Comment (mdimjasevic) [reply libs/libzauth/libzauth-c/Cargo.lock]: I just updated it to sodiumoxide 0.2.7 in https://github.com/wireapp/wire-server/pull/2327/commits/bd82723e17d504372c51b01bde1797db276f1cb3. I also added to the repo `Cargo.lock` under `libzauth` (note we've been discussing `libzauth-c/Cargo.lock` above). I have no clue if it should be tracked; if not, I'll stop tracking it.

**Comment by mdimjasevic** [comment]
@jschaul, did you say yesterday we might have to update `sodiumoxide` in some other place too?

**Comment by jschaul** [comment]
We're currently using libsodium 1.0.16 [here](https://github.com/zinfra/cailleach/blob/master/wire-server-private/ci/dockerfiles/ubuntu14-nginz-deps/Dockerfile#L4), there's libsodium 1.0.18 out there, @comawill do we need to /should we update this also?

**Review by comawill — CHANGES_REQUESTED** [review]
Good work, on updating this library and the dependencies.

Yes, we definitely should update the used libsodium, too.

Right now the build process does not execute the tests, we provide in the library, `cargo test` should be executed when building the debian files.

File: libs/libzauth/libzauth-c/src/lib.rs [hunk 2]
```diff
@@ -177,7 +176,6 @@ pub extern fn zauth_token_delete(t: *mut ZauthToken) {
 }
 
 #[repr(C)]
-#[no_mangle]
 #[derive(Clone, Copy, Debug)]
```
Comment (jschaul) [inline libs/libzauth/libzauth-c/src/lib.rs:179]: @comawill this is the place I'm least sure about my change: the no_mangle at the enum level led to warnings; but I'm not sure if removing them is okay to do here.

Comment (stephen-smith) [reply libs/libzauth/libzauth-c/src/lib.rs:179]: I think it's fine.  I don't think enum names or enum value names are part of the C ABI.

File: libs/libzauth/libzauth-c/Makefile [hunk 3]
```diff
@@ -43,7 +43,11 @@ uninstall:
 	rm -f $(PREFIX_INSTALL)/lib/libzauth.$(LIB_TYPE)
 	rm -f $(PREFIX_INSTALL)/lib/pkgconfig/libzauth.pc
 
-dist: build-release
+.PHONY: test
+test:
+	cd ../libzauth && cargo test
```
Comment (pcapriotti) [inline libs/libzauth/libzauth-c/Makefile:48]: Maybe add `--release` here?

File: libs/libzauth/libzauth-c/src/lib.rs (hunk 2 above)
Comment (pcapriotti) [reply libs/libzauth/libzauth-c/src/lib.rs:179]: I think `no_mangle` on structs and enums doesn't do anything, so it is safe to remove.

**Review by pcapriotti — APPROVED** [review]
LGTM

File: libs/libzauth/libzauth-c/Makefile (hunk 3 above)
Comment (jschaul) [reply libs/libzauth/libzauth-c/Makefile]: Done, thanks.

File: libs/libzauth/libzauth-c/src/lib.rs (hunk 2 above)
Comment (jschaul) [reply libs/libzauth/libzauth-c/src/lib.rs:179]: thanks for your reviews!

**Pull request merged**